pkg/decoder/payload.go

```diff
@@ -0,0 +1,53 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+package decoder
+
+import (
+	"github.com/DataDog/datadog-log-agent/pkg/message"
+)
+
+// PayloadContext can create new message.MessageOrigin from a msg
+// and be updated with content
+type PayloadContext interface {
```

Comment:
Context is a widely used word with a different meaning. I don't have a better name in mind, but it's worth thinking about an alternative (but feel free to stick with context if no better alternative)